dipy/tracking/utils.py

```diff
@@ -484,6 +489,45 @@ def target(streamlines, target_mask, affine, include=True):
             yield sl
 
 
+def near_roi(streamlines, target_mask, affine=None, tol=0):
+    """
+    Provide filtering criteria for a set of streamlines based on whether they
+    fall within a tolerance distance from an ROI
+
+    Parameters
+    ----------
+    streamlines: list
+        A sequence of streamlines. Each streamline should be a (N, 3) array,
+        where N is the length of the streamline.
+    target_mask: ndarray
+        A mask used as a target. Non-zero values are considered to be within
+        the target region.
+    affine: ndarray
+        Affine transformation from voxels to streamlines. Default: identity.
+    tol: float
+        Distance (in the units of the streamlines, usually mm). If any
+        coordinate in the streamline is within this distance from any voxel in
+        the ROI, the filtering criterion is set to True for this streamlin,
+        otherwise False. Default: 0
+
+    Returns
+    -------
+    Boolean array. Set to True in each streamline
+    """
+    if affine is None:
+        affine = np.eye(4)
+    roi_coords = np.array(np.where(target_mask)).T
+    x_roi_coords = apply_affine(affine, roi_coords)
+    out = np.zeros(len(streamlines), dtype=bool)
+    for ii, sl in enumerate(streamlines):
+        for coord in sl:
```

Comment:
Hey bro @arokem! This looks very inefficient. And it is important that we can have this roi-filter functions as fast as possible. So, can I suggest that you write your "first" Cython code to increase the speed of this function? It would be great way to increase your Cython expertise. Are you in for the challenge?